specs/FrameProcesssInfo.md

```diff
@@ -0,0 +1,503 @@
+Frame Process Info API
+===
+
+# Background
+Management of performance and overhead are some of the top concerns with 
+applications building more complex experiences using WebView2. We're seeing
+applications use multiple WebView2s, iframes, fencedframes, or other 
+embedding techniques for different aspects of their user's experience, and
+needing to prioritize or deprioritize the less important experiences depending
+on what the user is doing. This is similar to how the browser needs to 
+understand the impact of each tab in order to ensure that the background tabs 
+don't cause a major impact on the user's experience in the foreground tab.
+
+We provide the `GetProcessInfos` API for host applications to understand which
+processes are part of their WebView2s. That API provides enough information for
+a host application to understand the overall performance impact(memory, CPU 
+usage, etc.) of WebView2 on their application or the user's device, but it 
+doesn't provide the granularity needed for the host application to know which 
+part of WebView2 is consuming those resources.
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+# Description
+* We propose extending `CoreWebView2Environment` to include the 
+`GetProcessInfosWithDetails` API. This asynchronous call returns a 
+snapshot collection of `ProcessInfo`s corresponding to all currently
+running processes associated with this `ICoreWebView2Environment` 
+except for crashpad process. This provide the same list of `ProcessInfo`s
+as what's provided in `GetProcessInfos`. Plus, this also provide the list
+of associated `FrameInfo`s actively running(showing UI elements) in the 
+renderer process.
+
+* We propose to add the `AssociatedFrameInfo` API to provide a list of 
+`FrameInfo`s actively running(showing UI elements) in the asscociated 
+renderer process. 
+
+* We propose extending `CoreWebView2` and `CoreWebView2Frame` to include 
+the `FrameId` property. This property represents the unique identifier of
+the frame running in this `CoreWebView2` or `CoreWebView2Frame`.
+
+* We propose extending `CoreWebView2FrameInfo` to include `FrameId`, 
+`FrameKind` and `ParentFrameInfo` properties. `FrameId` is the same 
+kind of ID as with `FrameID` in `CoreWebView2` and `CoreWebView2Frame`. 
+`ParentFrameInfo` supports to retrieve a frame's direct parent. This 
+also can be used to build the frame tree represented by `FrameInfo`s.
+
+# Examples
+C++
+```c++
+void AppendFrameInfo(
+    wil::com_ptr<ICoreWebView2FrameInfo> frameInfo, std::wstringstream& result);
+wil::com_ptr<ICoreWebView2FrameInfo> GetAncestorFirstLevelFrameInfo(
+    wil::com_ptr<ICoreWebView2FrameInfo> frameInfo);
+wil::com_ptr<ICoreWebView2FrameInfo> GetAncestorMainFrameInfo(
+    wil::com_ptr<ICoreWebView2FrameInfo> frameInfo);
+std::wstring FrameKindToString(COREWEBVIEW2_FRAME_KIND kind);
+
+// Display renderer process info with details which includes the list of 

```

Comment:
right now "Details" specifically means AssociatedFrameInfos.  If we add more new optional details, will they also be populated by calls to GetProcessInfosWithDetails, or would there be a new method to opt into even more specific information? 
(I infer they'd be populated in calls to GetProcessInfosWithDetails, otherwise we'd name this method GetProcessInfosWithFramesInfo to avoid boxing out future method names.)

Reply:
As a tangible example for when this comment is discussed - We can theoretically have renderer processes for service workers that don't have an active frame.

Reply:
Considered we may provide more information for the processInfo. We'd like to extend `CoreWebView2ProcessInfo` to have more Associated...Info instead of adding more `CoreWebView2Environment.GetProcessInfosWith...Info` .

Reply:
Returning the sparsely populated Infos is confusing. Could the associated info be a getter method? As in: `CoreWebView2ProcessInfo.GetAssociatedFrameInfos()`?

Reply:
See above